### PR TITLE
Added upgrade-dd command to Registry manager to upgrade data dictionary

### PR DESCRIPTION
**Pull request by tdddblog**
## 🗒️ Summary
Added upgrade-dd command to Registry manager to upgrade data dictionary

**NOTE:** This branch includes changes from branch `versions` (https://github.com/NASA-PDS/registry-mgr/pull/46)

## ⚙️ Test Data and/or Report
* Run `registry-manager upgrade-dd -r`
* Check that `registry-dd` index was dropped and recreated and that data dictionary was loaded. You should see output similar to this:
```
[INFO] Deleting index registry-dd                                                                                                                      
[INFO] Creating index: registry-dd
[INFO] Schema: C:\tmp\registry-manager-4.5.0-SNAPSHOT\elastic\data-dic.json
[INFO] Shards: 1
[INFO] Replicas: 0
[INFO] Loading ES data file: C:\tmp\registry-manager-4.5.0-SNAPSHOT\elastic\data-dic-data.jar:dd.json
[INFO] Loaded 500 document(s)
[INFO] Loaded 1000 document(s)
[INFO] Loaded 1500 document(s)
[INFO] Loaded 2000 document(s)
[INFO] Loaded 2500 document(s)
[INFO] Loaded 3000 document(s)
[INFO] Loaded 3122 document(s)
```

## ♻️ Related Issues
https://github.com/NASA-PDS/registry/issues/43

### CODE REVIEW

File: src/main/java/gov/nasa/pds/registry/mgr/dao/RegistryDao.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        
+        Response resp = client.performRequest(req);
+        
+        // Check for Elasticsearch errors.
+        InputStream is = null;
+        InputStreamReader rd = null;
+        try
+        {
+            is = resp.getEntity().getContent();
+            rd = new InputStreamReader(is);
```
[the remark this answers is not in the document]
Comment (tdddblog) [reply src/main/java/gov/nasa/pds/registry/mgr/dao/RegistryDao.java:113]: ignore

File: src/main/java/gov/nasa/pds/registry/mgr/dao/resp/SettingsResponseParser.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+     * @throws Exception an exception
+     */
+    public IndexSettings parse(Response resp) throws Exception
+    {
+        if(resp == null) return null;
+        
+        InputStream is = resp.getEntity().getContent();
+        if(is == null) return null;
+        
+        JsonReader rd = new JsonReader(new InputStreamReader(is));
```
[the remark this answers is not in the document]
Comment (tdddblog) [reply src/main/java/gov/nasa/pds/registry/mgr/dao/resp/SettingsResponseParser.java:40]: ignore

File: src/main/java/gov/nasa/pds/registry/mgr/cmd/data/UpdateAltIdsCmd.java [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+    }
+
+    
+    private void updateIds(File file) throws Exception
+    {
+        BufferedReader rd = null;
+
+        try
+        {
+            rd = new BufferedReader(new FileReader(file));
```
[the remark this answers is not in the document]
Comment (tdddblog) [reply src/main/java/gov/nasa/pds/registry/mgr/cmd/data/UpdateAltIdsCmd.java:87]: ignore

File: src/main/java/gov/nasa/pds/registry/mgr/cmd/data/UpdateAltIdsCmd.java [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+        }
+
+        RegistryCfg cfg = new RegistryCfg();
+        cfg.url = cmdLine.getOptionValue("es", "http://localhost:9200");
+        cfg.indexName = cmdLine.getOptionValue("index", Constants.DEFAULT_REGISTRY_INDEX);
+        cfg.authFile = cmdLine.getOptionValue("auth");
+
+        String filePath = cmdLine.getOptionValue("file");
+        if(filePath == null) throw new Exception("Missing required parameter '-file'");
+        File file = new File(filePath);
```
[the remark this answers is not in the document]
Comment (tdddblog) [reply src/main/java/gov/nasa/pds/registry/mgr/cmd/data/UpdateAltIdsCmd.java:62]: ignore

File: src/main/java/gov/nasa/pds/registry/mgr/cmd/reg/CreateRegistryCmd.java [hunk 5]
```diff
@@ -173,42 +123,6 @@ private int parseInt(String str)
```
[the remark this answers is not in the document]
Comment (tdddblog) [reply src/main/java/gov/nasa/pds/registry/mgr/cmd/reg/CreateRegistryCmd.java]: fixed

File: src/main/java/gov/nasa/pds/registry/mgr/dao/RegistryDao.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (tdddblog) [reply src/main/java/gov/nasa/pds/registry/mgr/dao/RegistryDao.java:113]: ignore

File: src/main/java/gov/nasa/pds/registry/mgr/cmd/data/UpdateAltIdsCmd.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (tdddblog) [reply src/main/java/gov/nasa/pds/registry/mgr/cmd/data/UpdateAltIdsCmd.java:87]: ignore

File: src/main/java/gov/nasa/pds/registry/mgr/cmd/data/UpdateAltIdsCmd.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (tdddblog) [reply src/main/java/gov/nasa/pds/registry/mgr/cmd/data/UpdateAltIdsCmd.java:62]: ignore

File: src/main/java/gov/nasa/pds/registry/mgr/dao/resp/SettingsResponseParser.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (tdddblog) [reply src/main/java/gov/nasa/pds/registry/mgr/dao/resp/SettingsResponseParser.java:40]: ignore

**Comment by jordanpadams** [comment]
@jimmie can you confirm you ran this script successfully so we can go ahead with a merge?

**Comment by jimmie** [comment]
SBN-PSI has about 1.5M documents loaded so the script was timing out.

**Pull request merged**